src/main/java/io/playrconf/provider/GitProvider.java

```diff
@@ -0,0 +1,282 @@
+                        new ConfigException.ValidationProblem(
+                            config.getString("uri"),
+                            ex2.origin(),
+                            ex2.getMessage()
+                        )
+                    )
+                );
+            }
+        } catch (MalformedURLException | JGitInternalException | GitAPIException e) {
+            throw new ConfigException.BadPath("uri", e.getMessage());
+        } catch (IOException e) {
+            throw new RemoteConfException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Check if all required fields are set in the project config file.
+     * @param config Config file.
+     */
+    private void checkRequiredConfigFields(Config config) throws ConfigException {
+        if (!config.hasPath("mode") || config.getString("mode").isEmpty())
+            throw new ConfigException.Missing("mode");
+
+        if (!config.hasPath("uri") || config.getString("uri").isEmpty())
+            throw new ConfigException.Missing("uri");
+
+        if (!config.hasPath("branch") || config.getString("branch").isEmpty())
+            throw new ConfigException.Missing("branch");
+
+        if (!config.hasPath("filepath") || config.getString("filepath").isEmpty())
+            throw new ConfigException.Missing("filepath");
+
+        String mode = config.getString("mode").trim();
+        String repositoryURI = config.getString("uri").trim();
+
+        if (Objects.equals(mode, "user")) {
+            if (!repositoryURI.startsWith("http")) {
+                throw new ConfigException.BadValue("mode", String.format("Invalid repository URI for %s mode.", mode));
+            }
+
+            if (!config.hasPath("user.login") || config.getString("user.login").isEmpty()) {
+                throw new ConfigException.Missing("user.login");
+            } else if (!config.hasPath("user.password")) {
+                throw new ConfigException.Missing("user.password");
+            }
+        } else if (Objects.equals(mode, "ssh-rsa")) {
+            if (!repositoryURI.startsWith("git@")) {
+                throw new ConfigException.BadValue("mode", String.format("Invalid repository URI for %s mode.", mode));
+            }
+
+            if (!config.hasPath("ssh-rsa.private-key") || config.getString("ssh-rsa.private-key").isEmpty()) {
+                throw new ConfigException.Missing("ssh-rsa.private-key");
+            }
+        }
+    }
+
+    /**
+     * Clones the branch of Git repository.
+     *
+     * @param config Config file.
+     * @param repositoryURI Repository URI using HTTPS or SSH.
+     * @param mode Auth mode.
+     * @param branch Branch name.
+     * @return Repository.
+     */
+    private Repository cloneRepository(Config config, String repositoryURI, String mode, String branch) throws GitAPIException {
+        File repoDir = new File("play-rconf-git", String.format("%s-%s-%d", repositoryURI, branch, System.currentTimeMillis()));
```

Comment:
Is the directory deleted after ?

Reply:
@thibaultmeyer I've implemented the `Temp Folder` over `Files.createTempDirectory(...)` method.